pkg/dataobj/internal/dataset/value.go

```diff
@@ -126,6 +134,17 @@ func (v Value) String() string {
 	return v.Type().String()
 }
 
+// ByteSlice returns v's value as a byte slice. If v is not a string,
+// ByteSlice returns a byte slice of the form "VALUE_TYPE_T", where T is the
+// underlying type of v.
+func (v Value) ByteSlice() []byte {
+	if sp, ok := v.any.(stringptr); ok {
+		return unsafe.Slice(sp, v.num)
+	}
+	// TODO(benclive): Find out I should avoid this allocation
+	return []byte(v.Type().String())
```

Comment:
IMO I think ByteSlice should panic if v isn't a string or byte slice (per my previous comment); I'm not sure what the utility of it doing something for other Value types would be.